pkg/deploy/scaler/scale_test.go

```diff
@@ -0,0 +1,89 @@
+package scaler
+
+import (
+	"testing"
+	"time"
+
+	ktestclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client/testclient"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+
+	"github.com/openshift/origin/pkg/client/testclient"
+	deploytest "github.com/openshift/origin/pkg/deploy/api/test"
+)
+
+func TestScale(t *testing.T) {
+	tests := []struct {
+		testName               string
+		namespace              string
+		name                   string
+		count                  uint
+		preconditions          *kubectl.ScalePrecondition
+		retry, waitForReplicas *kubectl.RetryParams
+		oc                     *testclient.Fake
+		kc                     *ktestclient.Fake
+		expected               []string
+		kexpected              []string
+		expectedErr            error
+	}{
+		{
+			testName:  "simple scale",
+			namespace: "default",
+			name:      "foo",
+			count:     uint(3),
+			oc:        testclient.NewSimpleFake(deploytest.OkDeploymentConfig(1)),
+			kc:        ktestclient.NewSimpleFake(deploytest.MkDeploymentList(1)),
+			expected: []string{
+				"get-deploymentconfig",
+			},
+			kexpected: []string{
+				"get-replicationController",
+				"update-replicationController",
+			},
```

Comment:
I find this kind of expectation setting extremely hard to understand and maintain, but I'm not clear on the current preferences upstream. In the past I thought there was buy-in on using injected function mocks over the fake clients for code like this, but at least @derekwaynecarr seemed opposed. I don't mind letting it in here, just stating my opinion that I'd prefer a different style.